autometa/common/metagenome.py

```diff
@@ -357,11 +357,11 @@ def length_filter(self, out, cutoff=3000, force=False):
             cutoff value must be a positive real number
         FileExistsError
             filepath consisting of sequences that passed filter already exists
-
         """
-        if not isinstance(cutoff, numbers.Number) or isinstance(cutoff, bool):
-            # https://stackoverflow.com/a/4187220/13118765
-            raise TypeError(f"cutoff: {cutoff} must be a float or int")
+        try:
+            cutoff = float(cutoff)
+        except ValueError:
+            raise ValueError(f"Cutoff should be flaot of int, specified {cutoff}")
```

Comment:
This looks like we are replacing a more robust type check with a brute force conversion. I think we should leave the former